### ISSUE TITLE
copying using glob unexpected behaviour

### ISSUE THREAD

**Issue by samvloeberghs** [kevlened/copy-webpack-plugin#18]
Hi,

Assume I have the following dir structure
```
- src
-- app
--- picture.png
--- subdir
----- subpicture.png
--- stuff.txt
-- assets
--- stuff.txt
```

When I do a copy, in a new context of dist, using
```
new CopyWebpackPlugin([
      {
        from: 'src/assets',
        to: 'assets'
      }
])
```
it will copy all the files from `src/assets` to `dist/assets`.
But when I do a copy, in the same context of dist, using:
```
    new CopyWebpackPlugin([
      {
        from: 'src/assets',
        to: 'assets'
      },
      {
        from: 'src/app/**/*.png',
        to: 'app'
      }
    ])
```

it will copy all the files from `src/assets` to `dist/assets` **but** copy the png files in `src/app` to `dist/src/**/**.png`. *( mind the 'src' )*

This is very confusing behaviour, which I feel is not expected. So what's up? :)

Kind regards
Sam V.

**Comment by doxavore** [comment]
:+1: I'm experiencing this as well. I was expecting a behavior similar to gulp, wherein the base of the glob start will serve as the base for all copied files.

**Comment by iilei** [comment]
:+1: same here

**Comment by kevlened** [comment]
This is a bug in that the files should be copied to `dist/app/src/**/**.png` instead of `dist/src/**/**.png`. However, using the base of the glob as the base for all copied files is an enhancement.

**Comment by kevlened** [comment]
The bug portion of this issue is resolved in v2.0.1

**Comment by kevlened** [comment]
So, it's now possible (in v2.1.0) to change the base of the blob for a pattern by using a pattern-specific context.

    new CopyWebpackPlugin([{
      context: 'src/app',
      from: '**/*.png',
      to: 'app'
    }]);

That will copy all the `.png` files in src/app to app
